grafana-plugin/src/containers/PluginConfigPage/PluginConfigPage.tsx

```diff
@@ -1,54 +1,241 @@
+                          </Button>
+                          {isRecreatingServiceAccount && <LoadingPlaceholder text="" className={styles.spinner} />}
+                        </HorizontalGroup>
+                      </>
+                    ),
+                  },
+                ]),
+            {
+              ...(connectionStatus?.oncall_api_url?.ok ? { customIcon: 'check' as const } : { isTextIcon: true }),
+              startingElemPosition: '-6px',
+              expandedView: () => (
+                <>
+                  <Text strong>Let us know the backend URL for your OnCall API</Text>
+                  <Text type="secondary" className={styles.secondaryTitle}>
+                    OnCall backend must be reachable from your Grafana Installation. <br />
+                    You can run hobby, dev or production backend. See{' '}
+                    <a href={DOCS_ONCALL_OSS_INSTALL} target="_blank" rel="noreferrer">
+                      <Text type="link">here</Text>
+                    </a>{' '}
+                    how to get started.
+                  </Text>
+                  <form onSubmit={handleSubmit(onSubmit)}>
+                    <Controller
+                      name={'onCallApiUrl'}
+                      control={control}
+                      rules={{ required: 'OnCall API URL is required', validate: validateURL }}
+                      render={({ field }) => (
+                        <Field
+                          key={'Name'}
+                          label={'OnCall API URL'}
+                          invalid={Boolean(formState.errors.onCallApiUrl)}
+                          error={formState.errors.onCallApiUrl?.message}
+                        >
+                          <Input {...field} placeholder={'OnCall API URL'} data-testid="oncall-api-url-input" />
+                        </Field>
+                      )}
+                    />
+                    <HorizontalGroup>
+                      <Button
+                        type="submit"
+                        disabled={
+                          !formState.isValid ||
+                          !meta.enabled ||
+                          !(
+                            getIsExternalServiceAccountFeatureAvailable() || connectionStatus?.service_account_token?.ok
+                          )
+                        }
+                      >
+                        {isPluginConnected ? 'Reconnect' : 'Connect'}
+                      </Button>
+                      {isReinitializating && <LoadingPlaceholder text="" className={styles.spinner} />}
+                    </HorizontalGroup>
+                  </form>
+                </>
+              ),
+            },
+          ]}
+        />
+      </>
+    );
+  }
+);
+
+const PluginConfigAlert = observer(() => {
+  const {
+    pluginStore: { connectionStatus, isPluginConnected },
+  } = rootStore;
+
+  if (!connectionStatus) {
+    return null;
+  }
+  return isPluginConnected ? (
```

Comment:
IMO it's cleaner if you don't nest here and instead use an if on the previous line.

Reply:
Not sure if we can get better here, let's discuss on Slack